keras_core/layers/layer.py

```diff
@@ -1217,6 +1221,32 @@ def _get_call_context(self):
             self._clear_losses()
         return layer_call_ctx
 
+    def _set_save_spec(self, args, kwargs):
```

Comment:
This is currently causing some layer tests to fail. It's also partially redundant with another feature that we have, which is that we record the shapes of all tensor input arguments during `build()`.

You should be able to remove this and instead move the logic to the exporter. You can access `_build_shapes_dict` for the shapes of all input arguments (in dict format, e.g. with keys like `"input_shape"`). The dtypes are normally going to be `self.input_dtype`.

Something like:

```python
if len(shapes_dict) == 1:
   return TensorSpec(shape=shapes_dict.values()[0], dtype=self.input_dtype)
specs = {}
for key, value in shapes_dict.items():
   key = removesuffix(key, "_shape")
   specs[key] = TensorSpec(shape=value, dtype=self.input_dtype)
return specs
```

Reply:
Great suggestion, I've fixed this according to the flow you suggested, thanks!